bvm/ballerina-core/src/main/java/org/ballerinalang/core/model/types/BUnionType.java

```diff
@@ -91,8 +92,27 @@ public int getTag() {
 
     @Override
     public String toString() {
-        List<String> list = memberTypes.stream().map(BType::toString).collect(Collectors.toList());
-        return String.join("|", list);
+        String cachedToString;
+        if (resolving) {
+            if (this.typeName != null) {
+                return this.typeName;
+            } else {
+                return "...";
+            }
+        }
+        resolving = true;
+        StringBuilder sb = new StringBuilder();
+        int size = memberTypes != null ? memberTypes.size() : 0;
```

Comment:
Can `toString` get called when `memberTypes` is null?

Reply:
I have added `setMemberTypes` which returns if members list is null given to it. But looks like there is no way for this to happen now. But shall we keep this for now. I think if members are empty we don't need to create a member list at runtime. An example is below.
I would like to create an issue for this. What do you think?

```Ballerina
type EmptyUnion never|never;
EmptyUnion x = 3;
```

```LogTalk
ERROR [t2.bal:(2:16,2:17)] incompatible types: expected '', found 'int'
```